portfolio_optimization/cuFOLIO_portfolio_optimization/CVaR/03_advanced_topics.ipynb

```
@@ -33,6 +33,41 @@
+    "import subprocess\n",
+    "from IPython.display import display, HTML\n",
+    "\n",
+    "def check_gpu():\n",
+    "    try:\n",
+    "        output = subprocess.check_output(\"nvidia-smi\", shell=True).decode()\n",
+    "        display(HTML(f\"\"\"\n",
+    "        <div style=\"border:2px solid #4CAF50;padding:10px;border-radius:10px;background:#e8f5e9;\">\n",
+    "            <h3>✅ GPU is enabled</h3>\n",
+    "            <pre>{output.splitlines()[2]}</pre>\n",
+    "        </div>\n",
+    "        \"\"\"))\n",
```

Ruff (0.14.0)

`from cuopt.linear_programming.solver.solver_parameters import *` used; unable to detect undefined names

(F403)